core/src/main/java/com/scalar/db/storage/jdbc/JdbcService.java

```diff
@@ -88,10 +89,17 @@ public Scanner getScanner(Scan scan, Connection connection)
 
     TableMetadata tableMetadata = tableMetadataManager.getTableMetadata(scan);
 
-    SelectQuery selectQuery =
-        scan instanceof ScanAll
-            ? buildSelectQueryForScanAll((ScanAll) scan, tableMetadata)
-            : buildSelectQueryForScan(scan, tableMetadata);
+    SelectQuery selectQuery;
+    if (scan instanceof ScanAll) {
+      ScanAll scanAll = (ScanAll) scan;
+      if (ScalarDbUtils.isRelational(scan)) {
+        selectQuery = buildSelectQueryForRelationalScan(scanAll, tableMetadata);
```

Comment:
One of the major changes is starting from here. Since we decide to use `ScanAll` for relational scan, we do not have to change the interface of `DistributedStorage`, but just have to change query building for conditions and orderings.